src/modules/partition/core/DirFSRestrictLayout.h

```diff
@@ -0,0 +1,86 @@
+/* === This file is part of Calamares - <https://calamares.io> ===
+ *
+ *   SPDX-FileCopyrightText: 2018-2019 Collabora Ltd <arnaud.ferraris@collabora.com>
+ *   SPDX-FileCopyrightText: 2019 Adriaan de Groot <groot@kde.org>
+ *   SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ *   Calamares is Free Software: see the License-Identifier above.
+ *
+ */
+
+#ifndef DIRFSRESTRICTLAYOUT_H
+#define DIRFSRESTRICTLAYOUT_H
+
+#include "Config.h"
+
+// KPMcore
+#include <kpmcore/fs/filesystem.h>
+
+// Qt
+#include <QList>
+#include <QObject>
+#include <QVariantMap>
+
+class DirFSRestrictLayout
+{
+public:
+    struct DirFSRestrictEntry
+    {
+        QString dirPath;
+        QList< FileSystem::Type > dirAllowedFSTypes;
+        bool useOnlyWhenMountpoint;
```

Comment:
Is uninitialized with the default constructor you've written. If you in-class initialize it, you can `= default` the default constructor.

Reply:
Fixed using your suggestion, good catch and thanks for the tip about using `= default`!